### ISSUE TITLE
MA0002 - Reported where a comparer cannot be specified.

### ISSUE THREAD

**Issue by rcollette** [meziantou/Meziantou.Analyzer#427]
**Context**
Rule identifier: MA0002
Target Framework: .NET 7
Language version: 10.0

**Describe the bug**
MA0002 is reported on LINQ query syntax that uses comparisons such as order by and group by but, to the best of knowledge, such comparisons only work with the default comparator for the type, and therefore MA0002 should not be reported.

**To Reproduce**
Create a group by query similar to:

````c#
        IEnumerable<SoftwareProductEntitlements> result =
            from entitlement in appliedSoftwareEntitlement
            group entitlement by entitlement.SoftwareEntitlement.Product.Name
            into groupedEntitlements
            select ToSoftwareProductEntitlements(groupedEntitlements);
````

**Additional context**

**Comment by meziantou** [comment]
I think this is valid to report the diagnostic in this case by default. Whatever the way to use the `GroupBy` method, the rule should report it. If you want to disable MA0002 for the query operators, I'll add a new option to configure the rule in #430

**Comment by rcollette** [comment]
Wow, that was fast.  Thank you.
